.github/workflows/doc-builder.yml

```diff
@@ -6,14 +6,46 @@ on:
       
   # Allow the workflow to be triggered also manually.
   workflow_dispatch:
+  
+permissions:
+  id-token: write
+  contents: write
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.CI_AWS_ROLE_ARN }}
+          aws-region: us-west-2
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
       - uses: actions/setup-python@v2
         with:
           python-version: 3.x
+      - name: Setup .NET Core 6.0
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.0.x
       - run: pip install mkdocs-material==8.2.9
+      - run: pip install mkdocs-awesome-pages-plugin==2.8.0
+      - name: Restore dependencies
+        run: dotnet restore
+      - name: Build
+        run: dotnet build --no-restore
+      - name: Run Client Generator
+        run: |
+          cd ./src/AWS.Deploy.DocGenerator
+          dotnet run --project ./AWS.Deploy.DocGenerator.csproj
+      - name: Commit and Push next version
+        id: commit-push
+        run: |
+          git config --global user.email "github-aws-sdk-dotnet-automation@amazon.com"
+          git config --global user.name "aws-sdk-dotnet-automation"
+          git add site/
+          git commit -m "Update documentation generated by AWS.Deploy.DocGenerator"
```

Comment:
Is it problematic at all that this is running on main? Will we have to merge back to `dev` more often? Will this run afoul of any branch protection rules?

Reply:
This could be problematic if you try to modify the files on `dev` in a way that causes a merge conflict with the generator that is running on `main`. In general, you shouldn't do that since changes you do on the files directly will be overwritten. However, I was debating the approach here and considered creating a new workflow similar to the one that detects CDK template changes. If I do that, a PR check will fail if it you make changes to the recipes but you didn't run the generator. This will require the developer to take more steps as opposed to a handsoff approach. I'm open to doing things differently here. What are your thoughts?

Reply:
As discussed on slack, I added an additional PR check that runs the doc generator and checks for documentation changes. This way, the one on main would be a last resort if someone decides to ignore the PR check.